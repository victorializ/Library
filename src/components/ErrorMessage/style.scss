@import '../../assets/variables.scss';

.ui.segment {
    margin: 0;
}
.error-message {
    &__wrapper {
        width: 100%;
       @include column-flex-container;
    }
    width: 100%;  
    @include column-flex-container;
}

.ui.segment {
    &:last-child, &:first-child {
        margin-bottom: $space-xs;
    }
}