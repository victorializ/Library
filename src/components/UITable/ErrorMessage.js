import React from 'react';
import { Segment, Header } from 'semantic-ui-react';

import { colors } from '../../assets/semantic-colors';

const ErrorMessage = ({text}) => 
    <div className='center-horizontal'>
        <Segment 
            color={colors.red} 
            className='center-horizontal'
           >
            <Header className='error__text'>
                {text}
            </Header>
        </Segment>
    </div>

export { ErrorMessage };