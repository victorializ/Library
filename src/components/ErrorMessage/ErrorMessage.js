import React, { useState }  from 'react';

import { Segment, Header, Button } from 'semantic-ui-react';

import { constants } from '../../i18n';
import { colors } from '../../assets/semantic-colors';

import './style.scss';

const ErrorMessage = ({text}) => {
    const [ visible, setVisibility ] = useState(true);
    return (
        <div className='error-message__wrapper'>
        {
            visible && <Segment color={colors.red} className='error-message'>
                <Header>{text}</Header>
                {/*
                <Segment.Inline>
                    <Button basic color={colors.primary} onClick={() => setVisibility(false)}>
                        {constants.ok}
                    </Button>
                </Segment.Inline>
                */}
            </Segment>
        }
        </div>
    )
}
export { ErrorMessage };