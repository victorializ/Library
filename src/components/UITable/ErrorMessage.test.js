import React from 'react';
import { shallow } from 'enzyme';

import { ErrorMessage } from './ErrorMessage';

describe('ErrorMessage component', () => {
    it('renders correctly', () => {
        expect(shallow(<ErrorMessage />)).toMatchSnapshot();
    });   
    it('displays error message', () => {
        const text = 'text';
        const wrapper = shallow(<ErrorMessage text={text}/>);
        expect(wrapper.find('.error__text').render().text()).toEqual(text);
    });
});