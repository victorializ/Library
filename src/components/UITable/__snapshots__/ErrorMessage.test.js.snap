// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component renders correctly 1`] = `
<div
  className="center-horizontal"
>
  <Segment
    className="center-horizontal"
    color="red"
  >
    <Header
      className="error__text"
    />
  </Segment>
</div>
`;
